front/src/components/ui/Chip.tsx

```diff
@@ -1,24 +1,89 @@
-import { FC,useState } from "react";
+import { FC, useState } from "react";
 import ChipClick from '@mui/material/Chip';
-
+import Avatar from '@mui/material/Avatar';
+import InstagramIcon from '@mui/icons-material/Instagram';
+import XIcon from '@mui/icons-material/X';
+import FeedIcon from '@mui/icons-material/Feed';
+import MediumIcon from'../../../public/medium.svg'
+import TechCrunchIcon from '../../../public/techCrunch.svg'
+import TheGuardIcon from '../../../public/theGuardian.svg'
+import EuroParlamentIcon from '../../../public/euroParlament.svg'
+import nyTimesIcon from '../../../public/nytimes.svg';
+import googleDailyMixIcon from '../../../public/googleDailyMix.svg'
 interface ChipProps {
   label: string;
   onClick?: () => void;
+  onRemove?: () => void;
   isSelected?: boolean;
   type?: string;
+  chipType?:string;
+}
+const Icons={
+  inst: <InstagramIcon />,
+  x:<XIcon/>,
+  medium:<img src={MediumIcon}/>,
+  rss:<FeedIcon/>,
+  techCrunch:<img src={TechCrunchIcon}/>,
+  theGuardian:<img src={TheGuardIcon}/>,
+  europeanParliament:<img src={EuroParlamentIcon}/>,
+  nyTimes:<img src={nyTimesIcon}/>,
+  googleDailyMix:<img src={googleDailyMixIcon}/>
 }
 
-export const Chip: FC<ChipProps> = ({ label, onClick, isSelected, type='click' }) => {
+export const Chip: FC<ChipProps> = ({chipType, label, onClick, isSelected, type = 'click', onRemove }) => {
   const variant = isSelected ? 'filled' : 'outlined';
   const [isClicked, setIsClicked] = useState<boolean>(false);
+
   const handleClick = () => {
-    setIsClicked(!isClicked);
+    if (onRemove) {
```

Comment:
i don't like this if else if. Can you refactor somehow to make steps and handlers processing more clear